lib/kms.js

```diff
@@ -23,62 +22,59 @@ exports.createKeystore = async ({
   const config = {
     sequence: 0,
     controller: capabilityAgent.id,
-    // TODO: prefer using just controller when the invoker and delegator will be
-    //       the same
-    invoker: capabilityAgent.id,
-    delegator: capabilityAgent.id
   };
   if(referenceId) {
     config.referenceId = referenceId;
   }
-  const {httpsAgent} = brHttpsAgent;
   return KmsClient.createKeystore({
     url: `${kmsBaseUrl}/keystores`,
     config,
     httpsAgent
   });
 };
 
-// FIXME: this only works with an integrated KMS right now and that needs to be
-//        fixed in the future
-exports.updateKeystoreController = async ({id, controller} = {}) => {
-  const {httpsAgent} = brHttpsAgent;
-  const config = await KmsClient.getKeystore({id, httpsAgent});
-
-  const {invoker} = config;
-  if(!Array.isArray(invoker)) {
-    config.invoker = [invoker];
+exports.updateKeystoreController = async ({
+  controller, keystoreAgent, keystoreConfig
+} = {}) => {
+  // if the keystore config is not supplied, retrieve it now
+  if(!keystoreConfig) {
+    // FIXME: GET ID
+    // keystoreConfig = await KmsClient.getKeystore({id, httpsAgent});
```

Comment:
Seems this FIXME should be resolved prior to merging this PR.

Reply:
Turns out this is effectively a private helper API with exactly one usage where we have not need of looking up the config.  Factored this out.